// @aws-sdk/client-athena/dist-cjs/index.js
var __exportStar = (m, e) => { Object.assign(e, m); };
const { getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, Command, createAggregatedClient } = require("@smithy/core/client");
exports.$Command = Command;
exports.__Client = Client;
const { resolveRegionConfig } = require("@smithy/core/config");
const { resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveHttpAuthSchemeConfig, defaultAthenaHttpAuthSchemeParametersProvider } = require("./auth/httpAuthSchemeProvider");
const { getRuntimeConfig } = require("./runtimeConfig");
const { BatchGetNamedQuery$, BatchGetPreparedStatement$, BatchGetQueryExecution$, CancelCapacityReservation$, CreateCapacityReservation$, CreateDataCatalog$, CreateNamedQuery$, CreateNotebook$, CreatePreparedStatement$, CreatePresignedNotebookUrl$, CreateWorkGroup$, DeleteCapacityReservation$, DeleteDataCatalog$, DeleteNamedQuery$, DeleteNotebook$, DeletePreparedStatement$, DeleteWorkGroup$, ExportNotebook$, GetCalculationExecutionCode$, GetCalculationExecution$, GetCalculationExecutionStatus$, GetCapacityAssignmentConfiguration$, GetCapacityReservation$, GetDatabase$, GetDataCatalog$, GetNamedQuery$, GetNotebookMetadata$, GetPreparedStatement$, GetQueryExecution$, GetQueryResults$, GetQueryRuntimeStatistics$, GetResourceDashboard$, GetSession$, GetSessionEndpoint$, GetSessionStatus$, GetTableMetadata$, GetWorkGroup$, ImportNotebook$, ListApplicationDPUSizes$, ListCalculationExecutions$, ListCapacityReservations$, ListDatabases$, ListDataCatalogs$, ListEngineVersions$, ListExecutors$, ListNamedQueries$, ListNotebookMetadata$, ListNotebookSessions$, ListPreparedStatements$, ListQueryExecutions$, ListSessions$, ListTableMetadata$, ListTagsForResource$, ListWorkGroups$, PutCapacityAssignmentConfiguration$, StartCalculationExecution$, StartQueryExecution$, StartSession$, StopCalculationExecution$, StopQueryExecution$, TagResource$, TerminateSession$, UntagResource$, UpdateCapacityReservation$, UpdateDataCatalog$, UpdateNamedQuery$, UpdateNotebook$, UpdateNotebookMetadata$, UpdatePreparedStatement$, UpdateWorkGroup$ } = require("./schemas/schemas_0");
__exportStar(require("./schemas/schemas_0"), exports);
__exportStar(require("./models/errors"), exports);
const { AthenaServiceException } = require("./models/AthenaServiceException");
exports.AthenaServiceException = AthenaServiceException;

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "athena",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class AthenaClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultAthenaHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class BatchGetNamedQueryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "BatchGetNamedQuery", {})
    .n("AthenaClient", "BatchGetNamedQueryCommand")
    .sc(BatchGetNamedQuery$)
    .build() {
}

class BatchGetPreparedStatementCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "BatchGetPreparedStatement", {})
    .n("AthenaClient", "BatchGetPreparedStatementCommand")
    .sc(BatchGetPreparedStatement$)
    .build() {
}

class BatchGetQueryExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "BatchGetQueryExecution", {})
    .n("AthenaClient", "BatchGetQueryExecutionCommand")
    .sc(BatchGetQueryExecution$)
    .build() {
}

class CancelCapacityReservationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CancelCapacityReservation", {})
    .n("AthenaClient", "CancelCapacityReservationCommand")
    .sc(CancelCapacityReservation$)
    .build() {
}

class CreateCapacityReservationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateCapacityReservation", {})
    .n("AthenaClient", "CreateCapacityReservationCommand")
    .sc(CreateCapacityReservation$)
    .build() {
}

class CreateDataCatalogCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateDataCatalog", {})
    .n("AthenaClient", "CreateDataCatalogCommand")
    .sc(CreateDataCatalog$)
    .build() {
}

class CreateNamedQueryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateNamedQuery", {})
    .n("AthenaClient", "CreateNamedQueryCommand")
    .sc(CreateNamedQuery$)
    .build() {
}

class CreateNotebookCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateNotebook", {})
    .n("AthenaClient", "CreateNotebookCommand")
    .sc(CreateNotebook$)
    .build() {
}

class CreatePreparedStatementCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreatePreparedStatement", {})
    .n("AthenaClient", "CreatePreparedStatementCommand")
    .sc(CreatePreparedStatement$)
    .build() {
}

class CreatePresignedNotebookUrlCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreatePresignedNotebookUrl", {})
    .n("AthenaClient", "CreatePresignedNotebookUrlCommand")
    .sc(CreatePresignedNotebookUrl$)
    .build() {
}

class CreateWorkGroupCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateWorkGroup", {})
    .n("AthenaClient", "CreateWorkGroupCommand")
    .sc(CreateWorkGroup$)
    .build() {
}

class DeleteCapacityReservationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteCapacityReservation", {})
    .n("AthenaClient", "DeleteCapacityReservationCommand")
    .sc(DeleteCapacityReservation$)
    .build() {
}

class DeleteDataCatalogCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteDataCatalog", {})
    .n("AthenaClient", "DeleteDataCatalogCommand")
    .sc(DeleteDataCatalog$)
    .build() {
}

class DeleteNamedQueryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteNamedQuery", {})
    .n("AthenaClient", "DeleteNamedQueryCommand")
    .sc(DeleteNamedQuery$)
    .build() {
}

class DeleteNotebookCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteNotebook", {})
    .n("AthenaClient", "DeleteNotebookCommand")
    .sc(DeleteNotebook$)
    .build() {
}

class DeletePreparedStatementCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeletePreparedStatement", {})
    .n("AthenaClient", "DeletePreparedStatementCommand")
    .sc(DeletePreparedStatement$)
    .build() {
}

class DeleteWorkGroupCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteWorkGroup", {})
    .n("AthenaClient", "DeleteWorkGroupCommand")
    .sc(DeleteWorkGroup$)
    .build() {
}

class ExportNotebookCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ExportNotebook", {})
    .n("AthenaClient", "ExportNotebookCommand")
    .sc(ExportNotebook$)
    .build() {
}

class GetCalculationExecutionCodeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCalculationExecutionCode", {})
    .n("AthenaClient", "GetCalculationExecutionCodeCommand")
    .sc(GetCalculationExecutionCode$)
    .build() {
}

class GetCalculationExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCalculationExecution", {})
    .n("AthenaClient", "GetCalculationExecutionCommand")
    .sc(GetCalculationExecution$)
    .build() {
}

class GetCalculationExecutionStatusCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCalculationExecutionStatus", {})
    .n("AthenaClient", "GetCalculationExecutionStatusCommand")
    .sc(GetCalculationExecutionStatus$)
    .build() {
}

class GetCapacityAssignmentConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCapacityAssignmentConfiguration", {})
    .n("AthenaClient", "GetCapacityAssignmentConfigurationCommand")
    .sc(GetCapacityAssignmentConfiguration$)
    .build() {
}

class GetCapacityReservationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCapacityReservation", {})
    .n("AthenaClient", "GetCapacityReservationCommand")
    .sc(GetCapacityReservation$)
    .build() {
}

class GetDatabaseCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetDatabase", {})
    .n("AthenaClient", "GetDatabaseCommand")
    .sc(GetDatabase$)
    .build() {
}

class GetDataCatalogCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetDataCatalog", {})
    .n("AthenaClient", "GetDataCatalogCommand")
    .sc(GetDataCatalog$)
    .build() {
}

class GetNamedQueryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetNamedQuery", {})
    .n("AthenaClient", "GetNamedQueryCommand")
    .sc(GetNamedQuery$)
    .build() {
}

class GetNotebookMetadataCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetNotebookMetadata", {})
    .n("AthenaClient", "GetNotebookMetadataCommand")
    .sc(GetNotebookMetadata$)
    .build() {
}

class GetPreparedStatementCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetPreparedStatement", {})
    .n("AthenaClient", "GetPreparedStatementCommand")
    .sc(GetPreparedStatement$)
    .build() {
}

class GetQueryExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetQueryExecution", {})
    .n("AthenaClient", "GetQueryExecutionCommand")
    .sc(GetQueryExecution$)
    .build() {
}

class GetQueryResultsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetQueryResults", {})
    .n("AthenaClient", "GetQueryResultsCommand")
    .sc(GetQueryResults$)
    .build() {
}

class GetQueryRuntimeStatisticsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetQueryRuntimeStatistics", {})
    .n("AthenaClient", "GetQueryRuntimeStatisticsCommand")
    .sc(GetQueryRuntimeStatistics$)
    .build() {
}

class GetResourceDashboardCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetResourceDashboard", {})
    .n("AthenaClient", "GetResourceDashboardCommand")
    .sc(GetResourceDashboard$)
    .build() {
}

class GetSessionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetSession", {})
    .n("AthenaClient", "GetSessionCommand")
    .sc(GetSession$)
    .build() {
}

class GetSessionEndpointCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetSessionEndpoint", {})
    .n("AthenaClient", "GetSessionEndpointCommand")
    .sc(GetSessionEndpoint$)
    .build() {
}

class GetSessionStatusCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetSessionStatus", {})
    .n("AthenaClient", "GetSessionStatusCommand")
    .sc(GetSessionStatus$)
    .build() {
}

class GetTableMetadataCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetTableMetadata", {})
    .n("AthenaClient", "GetTableMetadataCommand")
    .sc(GetTableMetadata$)
    .build() {
}

class GetWorkGroupCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetWorkGroup", {})
    .n("AthenaClient", "GetWorkGroupCommand")
    .sc(GetWorkGroup$)
    .build() {
}

class ImportNotebookCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ImportNotebook", {})
    .n("AthenaClient", "ImportNotebookCommand")
    .sc(ImportNotebook$)
    .build() {
}

class ListApplicationDPUSizesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListApplicationDPUSizes", {})
    .n("AthenaClient", "ListApplicationDPUSizesCommand")
    .sc(ListApplicationDPUSizes$)
    .build() {
}

class ListCalculationExecutionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListCalculationExecutions", {})
    .n("AthenaClient", "ListCalculationExecutionsCommand")
    .sc(ListCalculationExecutions$)
    .build() {
}

class ListCapacityReservationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListCapacityReservations", {})
    .n("AthenaClient", "ListCapacityReservationsCommand")
    .sc(ListCapacityReservations$)
    .build() {
}

class ListDatabasesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListDatabases", {})
    .n("AthenaClient", "ListDatabasesCommand")
    .sc(ListDatabases$)
    .build() {
}

class ListDataCatalogsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListDataCatalogs", {})
    .n("AthenaClient", "ListDataCatalogsCommand")
    .sc(ListDataCatalogs$)
    .build() {
}

class ListEngineVersionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListEngineVersions", {})
    .n("AthenaClient", "ListEngineVersionsCommand")
    .sc(ListEngineVersions$)
    .build() {
}

class ListExecutorsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListExecutors", {})
    .n("AthenaClient", "ListExecutorsCommand")
    .sc(ListExecutors$)
    .build() {
}

class ListNamedQueriesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListNamedQueries", {})
    .n("AthenaClient", "ListNamedQueriesCommand")
    .sc(ListNamedQueries$)
    .build() {
}

class ListNotebookMetadataCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListNotebookMetadata", {})
    .n("AthenaClient", "ListNotebookMetadataCommand")
    .sc(ListNotebookMetadata$)
    .build() {
}

class ListNotebookSessionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListNotebookSessions", {})
    .n("AthenaClient", "ListNotebookSessionsCommand")
    .sc(ListNotebookSessions$)
    .build() {
}

class ListPreparedStatementsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListPreparedStatements", {})
    .n("AthenaClient", "ListPreparedStatementsCommand")
    .sc(ListPreparedStatements$)
    .build() {
}

class ListQueryExecutionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListQueryExecutions", {})
    .n("AthenaClient", "ListQueryExecutionsCommand")
    .sc(ListQueryExecutions$)
    .build() {
}

class ListSessionsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListSessions", {})
    .n("AthenaClient", "ListSessionsCommand")
    .sc(ListSessions$)
    .build() {
}

class ListTableMetadataCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListTableMetadata", {})
    .n("AthenaClient", "ListTableMetadataCommand")
    .sc(ListTableMetadata$)
    .build() {
}

class ListTagsForResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListTagsForResource", {})
    .n("AthenaClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class ListWorkGroupsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListWorkGroups", {})
    .n("AthenaClient", "ListWorkGroupsCommand")
    .sc(ListWorkGroups$)
    .build() {
}

class PutCapacityAssignmentConfigurationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "PutCapacityAssignmentConfiguration", {})
    .n("AthenaClient", "PutCapacityAssignmentConfigurationCommand")
    .sc(PutCapacityAssignmentConfiguration$)
    .build() {
}

class StartCalculationExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StartCalculationExecution", {})
    .n("AthenaClient", "StartCalculationExecutionCommand")
    .sc(StartCalculationExecution$)
    .build() {
}

class StartQueryExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StartQueryExecution", {})
    .n("AthenaClient", "StartQueryExecutionCommand")
    .sc(StartQueryExecution$)
    .build() {
}

class StartSessionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StartSession", {})
    .n("AthenaClient", "StartSessionCommand")
    .sc(StartSession$)
    .build() {
}

class StopCalculationExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StopCalculationExecution", {})
    .n("AthenaClient", "StopCalculationExecutionCommand")
    .sc(StopCalculationExecution$)
    .build() {
}

class StopQueryExecutionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StopQueryExecution", {})
    .n("AthenaClient", "StopQueryExecutionCommand")
    .sc(StopQueryExecution$)
    .build() {
}

class TagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "TagResource", {})
    .n("AthenaClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class TerminateSessionCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "TerminateSession", {})
    .n("AthenaClient", "TerminateSessionCommand")
    .sc(TerminateSession$)
    .build() {
}

class UntagResourceCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UntagResource", {})
    .n("AthenaClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateCapacityReservationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateCapacityReservation", {})
    .n("AthenaClient", "UpdateCapacityReservationCommand")
    .sc(UpdateCapacityReservation$)
    .build() {
}

class UpdateDataCatalogCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateDataCatalog", {})
    .n("AthenaClient", "UpdateDataCatalogCommand")
    .sc(UpdateDataCatalog$)
    .build() {
}

class UpdateNamedQueryCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateNamedQuery", {})
    .n("AthenaClient", "UpdateNamedQueryCommand")
    .sc(UpdateNamedQuery$)
    .build() {
}

class UpdateNotebookCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateNotebook", {})
    .n("AthenaClient", "UpdateNotebookCommand")
    .sc(UpdateNotebook$)
    .build() {
}

class UpdateNotebookMetadataCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateNotebookMetadata", {})
    .n("AthenaClient", "UpdateNotebookMetadataCommand")
    .sc(UpdateNotebookMetadata$)
    .build() {
}

class UpdatePreparedStatementCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdatePreparedStatement", {})
    .n("AthenaClient", "UpdatePreparedStatementCommand")
    .sc(UpdatePreparedStatement$)
    .build() {
}

class UpdateWorkGroupCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateWorkGroup", {})
    .n("AthenaClient", "UpdateWorkGroupCommand")
    .sc(UpdateWorkGroup$)
    .build() {
}

const paginateGetQueryResults = createPaginator(AthenaClient, GetQueryResultsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListApplicationDPUSizes = createPaginator(AthenaClient, ListApplicationDPUSizesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCalculationExecutions = createPaginator(AthenaClient, ListCalculationExecutionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCapacityReservations = createPaginator(AthenaClient, ListCapacityReservationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListDatabases = createPaginator(AthenaClient, ListDatabasesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListDataCatalogs = createPaginator(AthenaClient, ListDataCatalogsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListEngineVersions = createPaginator(AthenaClient, ListEngineVersionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListExecutors = createPaginator(AthenaClient, ListExecutorsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListNamedQueries = createPaginator(AthenaClient, ListNamedQueriesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListPreparedStatements = createPaginator(AthenaClient, ListPreparedStatementsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListQueryExecutions = createPaginator(AthenaClient, ListQueryExecutionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListSessions = createPaginator(AthenaClient, ListSessionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTableMetadata = createPaginator(AthenaClient, ListTableMetadataCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTagsForResource = createPaginator(AthenaClient, ListTagsForResourceCommand, "NextToken", "NextToken", "MaxResults");

const paginateListWorkGroups = createPaginator(AthenaClient, ListWorkGroupsCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    BatchGetNamedQueryCommand,
    BatchGetPreparedStatementCommand,
    BatchGetQueryExecutionCommand,
    CancelCapacityReservationCommand,
    CreateCapacityReservationCommand,
    CreateDataCatalogCommand,
    CreateNamedQueryCommand,
    CreateNotebookCommand,
    CreatePreparedStatementCommand,
    CreatePresignedNotebookUrlCommand,
    CreateWorkGroupCommand,
    DeleteCapacityReservationCommand,
    DeleteDataCatalogCommand,
    DeleteNamedQueryCommand,
    DeleteNotebookCommand,
    DeletePreparedStatementCommand,
    DeleteWorkGroupCommand,
    ExportNotebookCommand,
    GetCalculationExecutionCommand,
    GetCalculationExecutionCodeCommand,
    GetCalculationExecutionStatusCommand,
    GetCapacityAssignmentConfigurationCommand,
    GetCapacityReservationCommand,
    GetDatabaseCommand,
    GetDataCatalogCommand,
    GetNamedQueryCommand,
    GetNotebookMetadataCommand,
    GetPreparedStatementCommand,
    GetQueryExecutionCommand,
    GetQueryResultsCommand,
    GetQueryRuntimeStatisticsCommand,
    GetResourceDashboardCommand,
    GetSessionCommand,
    GetSessionEndpointCommand,
    GetSessionStatusCommand,
    GetTableMetadataCommand,
    GetWorkGroupCommand,
    ImportNotebookCommand,
    ListApplicationDPUSizesCommand,
    ListCalculationExecutionsCommand,
    ListCapacityReservationsCommand,
    ListDatabasesCommand,
    ListDataCatalogsCommand,
    ListEngineVersionsCommand,
    ListExecutorsCommand,
    ListNamedQueriesCommand,
    ListNotebookMetadataCommand,
    ListNotebookSessionsCommand,
    ListPreparedStatementsCommand,
    ListQueryExecutionsCommand,
    ListSessionsCommand,
    ListTableMetadataCommand,
    ListTagsForResourceCommand,
    ListWorkGroupsCommand,
    PutCapacityAssignmentConfigurationCommand,
    StartCalculationExecutionCommand,
    StartQueryExecutionCommand,
    StartSessionCommand,
    StopCalculationExecutionCommand,
    StopQueryExecutionCommand,
    TagResourceCommand,
    TerminateSessionCommand,
    UntagResourceCommand,
    UpdateCapacityReservationCommand,
    UpdateDataCatalogCommand,
    UpdateNamedQueryCommand,
    UpdateNotebookCommand,
    UpdateNotebookMetadataCommand,
    UpdatePreparedStatementCommand,
    UpdateWorkGroupCommand,
};
const paginators = {
    paginateGetQueryResults,
    paginateListApplicationDPUSizes,
    paginateListCalculationExecutions,
    paginateListCapacityReservations,
    paginateListDatabases,
    paginateListDataCatalogs,
    paginateListEngineVersions,
    paginateListExecutors,
    paginateListNamedQueries,
    paginateListPreparedStatements,
    paginateListQueryExecutions,
    paginateListSessions,
    paginateListTableMetadata,
    paginateListTagsForResource,
    paginateListWorkGroups,
};
class Athena extends AthenaClient {
}
createAggregatedClient(commands, Athena, { paginators });

const S3AclOption = {
    BUCKET_OWNER_FULL_CONTROL: "BUCKET_OWNER_FULL_CONTROL",
};
const AuthenticationType = {
    DIRECTORY_IDENTITY: "DIRECTORY_IDENTITY",
};
const EncryptionOption = {
    CSE_KMS: "CSE_KMS",
    SSE_KMS: "SSE_KMS",
    SSE_S3: "SSE_S3",
};
const StatementType = {
    DDL: "DDL",
    DML: "DML",
    UTILITY: "UTILITY",
};
const QueryExecutionState = {
    CANCELLED: "CANCELLED",
    FAILED: "FAILED",
    QUEUED: "QUEUED",
    RUNNING: "RUNNING",
    SUCCEEDED: "SUCCEEDED",
};
const DataCatalogType = {
    FEDERATED: "FEDERATED",
    GLUE: "GLUE",
    HIVE: "HIVE",
    LAMBDA: "LAMBDA",
};
const ConnectionType = {
    BIGQUERY: "BIGQUERY",
    CMDB: "CMDB",
    DATALAKEGEN2: "DATALAKEGEN2",
    DB2: "DB2",
    DB2AS400: "DB2AS400",
    DOCUMENTDB: "DOCUMENTDB",
    DYNAMODB: "DYNAMODB",
    GOOGLECLOUDSTORAGE: "GOOGLECLOUDSTORAGE",
    HBASE: "HBASE",
    MYSQL: "MYSQL",
    OPENSEARCH: "OPENSEARCH",
    ORACLE: "ORACLE",
    POSTGRESQL: "POSTGRESQL",
    REDSHIFT: "REDSHIFT",
    SAPHANA: "SAPHANA",
    SNOWFLAKE: "SNOWFLAKE",
    SQLSERVER: "SQLSERVER",
    SYNAPSE: "SYNAPSE",
    TIMESTREAM: "TIMESTREAM",
    TPCDS: "TPCDS",
};
const DataCatalogStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_FAILED_CLEANUP_COMPLETE: "CREATE_FAILED_CLEANUP_COMPLETE",
    CREATE_FAILED_CLEANUP_FAILED: "CREATE_FAILED_CLEANUP_FAILED",
    CREATE_FAILED_CLEANUP_IN_PROGRESS: "CREATE_FAILED_CLEANUP_IN_PROGRESS",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
const ThrottleReason = {
    CONCURRENT_QUERY_LIMIT_EXCEEDED: "CONCURRENT_QUERY_LIMIT_EXCEEDED",
};
const NotebookType = {
    IPYNB: "IPYNB",
};
const CalculationExecutionState = {
    CANCELED: "CANCELED",
    CANCELING: "CANCELING",
    COMPLETED: "COMPLETED",
    CREATED: "CREATED",
    CREATING: "CREATING",
    FAILED: "FAILED",
    QUEUED: "QUEUED",
    RUNNING: "RUNNING",
};
const CapacityAllocationStatus = {
    FAILED: "FAILED",
    PENDING: "PENDING",
    SUCCEEDED: "SUCCEEDED",
};
const CapacityReservationStatus = {
    ACTIVE: "ACTIVE",
    CANCELLED: "CANCELLED",
    CANCELLING: "CANCELLING",
    FAILED: "FAILED",
    PENDING: "PENDING",
    UPDATE_PENDING: "UPDATE_PENDING",
};
const QueryResultType = {
    DATA_MANIFEST: "DATA_MANIFEST",
    DATA_ROWS: "DATA_ROWS",
};
const ColumnNullable = {
    NOT_NULL: "NOT_NULL",
    NULLABLE: "NULLABLE",
    UNKNOWN: "UNKNOWN",
};
const SessionState = {
    BUSY: "BUSY",
    CREATED: "CREATED",
    CREATING: "CREATING",
    DEGRADED: "DEGRADED",
    FAILED: "FAILED",
    IDLE: "IDLE",
    TERMINATED: "TERMINATED",
    TERMINATING: "TERMINATING",
};
const WorkGroupState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ExecutorState = {
    CREATED: "CREATED",
    CREATING: "CREATING",
    FAILED: "FAILED",
    REGISTERED: "REGISTERED",
    TERMINATED: "TERMINATED",
    TERMINATING: "TERMINATING",
};
const ExecutorType = {
    COORDINATOR: "COORDINATOR",
    GATEWAY: "GATEWAY",
    WORKER: "WORKER",
};

exports.Athena = Athena;
exports.AthenaClient = AthenaClient;
exports.AuthenticationType = AuthenticationType;
exports.BatchGetNamedQueryCommand = BatchGetNamedQueryCommand;
exports.BatchGetPreparedStatementCommand = BatchGetPreparedStatementCommand;
exports.BatchGetQueryExecutionCommand = BatchGetQueryExecutionCommand;
exports.CalculationExecutionState = CalculationExecutionState;
exports.CancelCapacityReservationCommand = CancelCapacityReservationCommand;
exports.CapacityAllocationStatus = CapacityAllocationStatus;
exports.CapacityReservationStatus = CapacityReservationStatus;
exports.ColumnNullable = ColumnNullable;
exports.ConnectionType = ConnectionType;
exports.CreateCapacityReservationCommand = CreateCapacityReservationCommand;
exports.CreateDataCatalogCommand = CreateDataCatalogCommand;
exports.CreateNamedQueryCommand = CreateNamedQueryCommand;
exports.CreateNotebookCommand = CreateNotebookCommand;
exports.CreatePreparedStatementCommand = CreatePreparedStatementCommand;
exports.CreatePresignedNotebookUrlCommand = CreatePresignedNotebookUrlCommand;
exports.CreateWorkGroupCommand = CreateWorkGroupCommand;
exports.DataCatalogStatus = DataCatalogStatus;
exports.DataCatalogType = DataCatalogType;
exports.DeleteCapacityReservationCommand = DeleteCapacityReservationCommand;
exports.DeleteDataCatalogCommand = DeleteDataCatalogCommand;
exports.DeleteNamedQueryCommand = DeleteNamedQueryCommand;
exports.DeleteNotebookCommand = DeleteNotebookCommand;
exports.DeletePreparedStatementCommand = DeletePreparedStatementCommand;
exports.DeleteWorkGroupCommand = DeleteWorkGroupCommand;
exports.EncryptionOption = EncryptionOption;
exports.ExecutorState = ExecutorState;
exports.ExecutorType = ExecutorType;
exports.ExportNotebookCommand = ExportNotebookCommand;
exports.GetCalculationExecutionCodeCommand = GetCalculationExecutionCodeCommand;
exports.GetCalculationExecutionCommand = GetCalculationExecutionCommand;
exports.GetCalculationExecutionStatusCommand = GetCalculationExecutionStatusCommand;
exports.GetCapacityAssignmentConfigurationCommand = GetCapacityAssignmentConfigurationCommand;
exports.GetCapacityReservationCommand = GetCapacityReservationCommand;
exports.GetDataCatalogCommand = GetDataCatalogCommand;
exports.GetDatabaseCommand = GetDatabaseCommand;
exports.GetNamedQueryCommand = GetNamedQueryCommand;
exports.GetNotebookMetadataCommand = GetNotebookMetadataCommand;
exports.GetPreparedStatementCommand = GetPreparedStatementCommand;
exports.GetQueryExecutionCommand = GetQueryExecutionCommand;
exports.GetQueryResultsCommand = GetQueryResultsCommand;
exports.GetQueryRuntimeStatisticsCommand = GetQueryRuntimeStatisticsCommand;
exports.GetResourceDashboardCommand = GetResourceDashboardCommand;
exports.GetSessionCommand = GetSessionCommand;
exports.GetSessionEndpointCommand = GetSessionEndpointCommand;
exports.GetSessionStatusCommand = GetSessionStatusCommand;
exports.GetTableMetadataCommand = GetTableMetadataCommand;
exports.GetWorkGroupCommand = GetWorkGroupCommand;
exports.ImportNotebookCommand = ImportNotebookCommand;
exports.ListApplicationDPUSizesCommand = ListApplicationDPUSizesCommand;
exports.ListCalculationExecutionsCommand = ListCalculationExecutionsCommand;
exports.ListCapacityReservationsCommand = ListCapacityReservationsCommand;
exports.ListDataCatalogsCommand = ListDataCatalogsCommand;
exports.ListDatabasesCommand = ListDatabasesCommand;
exports.ListEngineVersionsCommand = ListEngineVersionsCommand;
exports.ListExecutorsCommand = ListExecutorsCommand;
exports.ListNamedQueriesCommand = ListNamedQueriesCommand;
exports.ListNotebookMetadataCommand = ListNotebookMetadataCommand;
exports.ListNotebookSessionsCommand = ListNotebookSessionsCommand;
exports.ListPreparedStatementsCommand = ListPreparedStatementsCommand;
exports.ListQueryExecutionsCommand = ListQueryExecutionsCommand;
exports.ListSessionsCommand = ListSessionsCommand;
exports.ListTableMetadataCommand = ListTableMetadataCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListWorkGroupsCommand = ListWorkGroupsCommand;
exports.NotebookType = NotebookType;
exports.PutCapacityAssignmentConfigurationCommand = PutCapacityAssignmentConfigurationCommand;
exports.QueryExecutionState = QueryExecutionState;
exports.QueryResultType = QueryResultType;
exports.S3AclOption = S3AclOption;
exports.SessionState = SessionState;
exports.StartCalculationExecutionCommand = StartCalculationExecutionCommand;
exports.StartQueryExecutionCommand = StartQueryExecutionCommand;
exports.StartSessionCommand = StartSessionCommand;
exports.StatementType = StatementType;
exports.StopCalculationExecutionCommand = StopCalculationExecutionCommand;
exports.StopQueryExecutionCommand = StopQueryExecutionCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.TerminateSessionCommand = TerminateSessionCommand;
exports.ThrottleReason = ThrottleReason;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateCapacityReservationCommand = UpdateCapacityReservationCommand;
exports.UpdateDataCatalogCommand = UpdateDataCatalogCommand;
exports.UpdateNamedQueryCommand = UpdateNamedQueryCommand;
exports.UpdateNotebookCommand = UpdateNotebookCommand;
exports.UpdateNotebookMetadataCommand = UpdateNotebookMetadataCommand;
exports.UpdatePreparedStatementCommand = UpdatePreparedStatementCommand;
exports.UpdateWorkGroupCommand = UpdateWorkGroupCommand;
exports.WorkGroupState = WorkGroupState;
exports.paginateGetQueryResults = paginateGetQueryResults;
exports.paginateListApplicationDPUSizes = paginateListApplicationDPUSizes;
exports.paginateListCalculationExecutions = paginateListCalculationExecutions;
exports.paginateListCapacityReservations = paginateListCapacityReservations;
exports.paginateListDataCatalogs = paginateListDataCatalogs;
exports.paginateListDatabases = paginateListDatabases;
exports.paginateListEngineVersions = paginateListEngineVersions;
exports.paginateListExecutors = paginateListExecutors;
exports.paginateListNamedQueries = paginateListNamedQueries;
exports.paginateListPreparedStatements = paginateListPreparedStatements;
exports.paginateListQueryExecutions = paginateListQueryExecutions;
exports.paginateListSessions = paginateListSessions;
exports.paginateListTableMetadata = paginateListTableMetadata;
exports.paginateListTagsForResource = paginateListTagsForResource;
exports.paginateListWorkGroups = paginateListWorkGroups;
